appinventor/components/src/com/google/appinventor/components/runtime/BluetoothConnectionBase.java

```diff
@@ -193,7 +194,11 @@ public final void Disconnect() {
    */
   @SimpleProperty(category = PropertyCategory.BEHAVIOR)
   public final boolean IsConnected() {
-    return (connectedBluetoothSocket != null);
+    if (sdkLevel >= SdkLevel.LEVEL_ICE_CREAM_SANDWICH) {
+      return (connectedBluetoothSocket != null && BluetoothReflection.isBluetoothSocketConnected(connectedBluetoothSocket));
+    }
+    else
```

Comment:
`else` case should still have curly braces even for a single statement ([Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)).